lib/index.js

```diff
@@ -83,6 +86,20 @@ const downloadFromInfoCallback = (stream, info, options) => {
   };
 
   let req;
+  let shouldEnd = true;
+  stream.on('set-pipe-and-events', () => {
```

Comment:
make this a function `const pipeAndSetEvents = ...` rather than an event. this event wouldn't be used by users of this library, so it's not necessary to expose it.